core/src/execution/client/verifiable_api.rs

```diff
@@ -0,0 +1,541 @@
+    }
+
+    async fn get_logs(&self, filter: &Filter) -> Result<Vec<Log>> {
+        let LogsResponse {
+            logs,
+            receipt_proofs,
+        } = self.api.get_logs(filter).await?;
+
+        self.verify_logs_and_receipts(&logs, receipt_proofs).await?;
+
+        Ok(logs)
+    }
+
+    async fn get_filter_changes(&self, filter_id: U256) -> Result<FilterChanges> {
+        let filter_changes = self.api.get_filter_changes(filter_id).await?;
+
+        Ok(match filter_changes {
+            FilterChangesResponse::Hashes(hashes) => FilterChanges::Hashes(hashes),
+            FilterChangesResponse::Logs(FilterLogsResponse {
+                logs,
+                receipt_proofs,
+            }) => {
+                self.verify_logs_and_receipts(&logs, receipt_proofs).await?;
+                FilterChanges::Logs(logs)
+            }
+        })
+    }
+
+    async fn get_filter_logs(&self, filter_id: U256) -> Result<Vec<Log>> {
+        let FilterLogsResponse {
+            logs,
+            receipt_proofs,
+        } = self.api.get_filter_logs(filter_id).await?;
+
+        self.verify_logs_and_receipts(&logs, receipt_proofs).await?;
+
+        Ok(logs)
+    }
+
+    async fn create_access_list(
+        &self,
+        tx: &N::TransactionRequest,
+        block_id: Option<BlockId>,
+    ) -> Result<HashMap<Address, Account>> {
+        let block_id = block_id.unwrap_or_default();
+        let tag = BlockTag::try_from(block_id)?;
+        let block = self
+            .state
+            .get_block(tag)
+            .await
+            .ok_or(ExecutionError::BlockNotFound(tag))?;
+        let block_id = BlockId::Number(block.header().number().into());
+
+        let AccessListResponse { accounts } = self
+            .api
+            .create_access_list(tx.clone(), Some(block_id))
+            .await?;
+
+        for (address, account) in &accounts {
+            self.verify_account(*address, account, &block)?;
+        }
+
+        Ok(accounts)
+    }
+
+    async fn chain_id(&self) -> Result<u64> {
+        let ChainIdResponse { chain_id } = self.api.chain_id().await?;
+        Ok(chain_id)
+    }
+
+    async fn get_block(
+        &self,
+        block_id: BlockId,
+        full_tx: bool,
+    ) -> Result<Option<N::BlockResponse>> {
+        self.api.get_block(block_id, full_tx).await
+    }
+
+    async fn get_block_receipts(
+        &self,
+        block_id: BlockId,
+    ) -> Result<Option<Vec<N::ReceiptResponse>>> {
+        self.api.get_block_receipts(block_id).await
+    }
+
+    async fn send_raw_transaction(&self, bytes: &[u8]) -> Result<B256> {
+        let SendRawTxResponse { hash } = self.api.send_raw_transaction(bytes).await?;
+        Ok(hash)
+    }
+
+    async fn new_filter(&self, filter: &Filter) -> Result<U256> {
+        let NewFilterResponse { id, .. } = self.api.new_filter(filter).await?;
+        Ok(id)
+    }
+
+    async fn new_block_filter(&self) -> Result<U256> {
```

Comment:
unused?

Reply:
So, the filter is created on the underlying RPC, and its filter ID is stored in Helios and returned to the user. However, when the user requests filter changes for this filter ID, the response is constructed from Helios's state without querying the underlying RPC. This means we can likely eliminate the initial call as well, generating the filter ID on helios side itself.

Reply:
Yeah lets do that

Reply:
So we should get rid of `new_block_filter` from both RPC and API altogether since it will not be used by Helios?

Reply:
Yeah I think if we have unused methods it makes sense to cut them unless you see a reason to keep them?

Reply:
I think we should keep it in the Verifiable-API for completeness, even if Helios doesn't use it.  

The `POST /eth/v1/filter` endpoint allows requests with a `kind` value of `Logs`, `NewBlocks`, or `NewPendingTransactions`, effectively replacing the `eth_newFilter`, `eth_newBlockFilter`, and `eth_newPendingTransactionFilter` RPC methods.  

Removing `NewBlocks` would make the implementation feel incomplete.